### PR TITLE
Add version number to composer metadata logs

### PR DESCRIPTION
**Pull request by j316chuck**
# What does this PR do?
Add version number to composer metadata logs

# What issue(s) does this change relate to?
Users have difficulty debugging which version of composer was running. It should be more clear now after 0.16.2. 

# Testing
Added unit test

### CODE REVIEW

File: composer/loggers/mosaicml_logger.py [hunk 1]
```diff
@@ -88,6 +89,8 @@ def log_metrics(self, metrics: Dict[str, Any], step: Optional[int] = None) -> No
         self._log_metadata(metrics)
 
     def after_load(self, state: State, logger: Logger) -> None:
+        # Log composer version
+        self._log_metadata({'composer_version': composer_version})
```
Comment (mvpatel2000) [inline composer/loggers/mosaicml_logger.py:93]: I don't think this should be here. It should probably just be directly in `composer.__init__`?

Comment (j316chuck) [reply composer/loggers/mosaicml_logger.py]: Thanks for the suggestion! Fixed 👍

File: composer/trainer/trainer.py [hunk 2]
```diff
@@ -1159,6 +1160,9 @@ def __init__(
             self.local_hparams = extract_hparams(locals())
             self.logger.log_hyperparameters(self.local_hparams)
 
+        # Log composer version.
+        self.logger.log_hyperparameters({'composer_version': composer_version})
```
Comment (dakinggg) [inline composer/trainer/trainer.py:1164]: should be able to just use `composer.__version__`?

Comment (dakinggg) [inline composer/trainer/trainer.py:1164]: if we're going to log this, lets also log the git commit if it exists? there is a utility func for this already in `collect_env.py`

Comment (dakinggg) [reply composer/trainer/trainer.py:1164]: but also might as well use the `get_composer_version` utility func in `collect_env.py`

Comment (j316chuck) [reply composer/trainer/trainer.py]: fixed

Comment (j316chuck) [reply composer/trainer/trainer.py]: fixd

**Review by dakinggg — APPROVED** [review]

File: composer/trainer/trainer.py [hunk 3]
```diff
@@ -1159,6 +1159,11 @@ def __init__(
             self.local_hparams = extract_hparams(locals())
             self.logger.log_hyperparameters(self.local_hparams)
 
+        # Log composer version.
+        composer_env_dict = get_composer_env_dict()
+        self.logger.log_hyperparameters({'composer_version': composer_env_dict['composer_version']})
+        self.logger.log_hyperparameters({'composer_commit_hash': str(composer_env_dict['composer_commit_hash'])})
```
Comment (mvpatel2000) [inline composer/trainer/trainer.py:1165]:
```suggestion
        # Log composer version
        composer_env_dict = get_composer_env_dict()
        self.logger.log_hyperparameters({'composer_version': composer_env_dict['composer_version']})
        self.logger.log_hyperparameters({'composer_commit_hash': str(composer_env_dict['composer_commit_hash'])})
```

**Review by eracah — APPROVED** [review]
LGTM

**Pull request merged**